src/tests/EventHistogramWriterTests.cpp

```diff
@@ -0,0 +1,547 @@
+    "edges": [2, 3, 4, 5, 6],
+    "dataset_name": "some_detector"
+  })""");
+}
+
+TEST_F(EventHistogramWriter, DimensionWithoutSizeThrows) {
+  auto Json = createTestDimensionJson();
+  Json.erase("size");
+  ASSERT_THROW(Dimension<double>::createFromJson(Json), UnexpectedJsonInput);
+}
+
+TEST_F(EventHistogramWriter, DimensionWithoutLabelThrows) {
+  auto Json = createTestDimensionJson();
+  Json.erase("label");
+  ASSERT_THROW(Dimension<double>::createFromJson(Json), UnexpectedJsonInput);
+}
+
+TEST_F(EventHistogramWriter, DimensionWithoutUnitThrows) {
+  auto Json = createTestDimensionJson();
+  Json.erase("unit");
+  ASSERT_THROW(Dimension<double>::createFromJson(Json), UnexpectedJsonInput);
+}
+
+TEST_F(EventHistogramWriter, DimensionCreatedFromValidInput) {
+  auto Json = createTestDimensionJson();
+  auto Dim = Dimension<double>::createFromJson(Json);
+  ASSERT_EQ(Dim.getSize(), 4u);
+  ASSERT_EQ(Dim.getLabel(), "Velocity");
+  ASSERT_EQ(Dim.getUnit(), "m/s");
+  ASSERT_EQ(Dim.getEdges().size(), 5u);
+}
+
+TEST_F(EventHistogramWriter, ShapeFromJsonThrowsIfInvalidInput) {
+  // The value of "shape" should be an array, so this should throw:
+  auto Json = json::parse(R"""({ "shape": {} })""");
+  ASSERT_THROW(Shape<double>::createFromJson(Json), UnexpectedJsonInput);
+}
+
+json createTestShapeJson() {
+  return json::parse(R""([
+    {
+      "size": 4,
+      "label": "Position",
+      "unit": "mm",
+      "edges": [2, 3, 4, 5, 6],
+      "dataset_name": "x_detector"
+    },
+    {
+      "size": 6,
+      "label": "Position",
+      "unit": "mm",
+      "edges": [-3, -2, -1, 0, 1, 2, 3],
+      "dataset_name": "y_detector"
+    },
+    {
+      "size": 3,
+      "label": "Time",
+      "unit": "ns",
+      "edges": [0, 2, 4, 6],
+      "dataset_name": "time_binning"
+    }
+  ])"");
+}
+
+TEST_F(EventHistogramWriter, ShapeCreatedFromValidInput) {
+  auto Json = createTestShapeJson();
+  auto TheShape = Shape<double>::createFromJson(Json);
+  ASSERT_EQ(TheShape.getNDIM(), 3u);
+}
+
+TEST_F(EventHistogramWriter, SliceNoOverlap) {
+  auto A = Slice::fromOffsetsSizes({0}, {2});
+  auto B = Slice::fromOffsetsSizes({2}, {2});
+  ASSERT_FALSE(A.doesOverlap(B));
+  ASSERT_FALSE(B.doesOverlap(A));
+  A = Slice::fromOffsetsSizes({0, 6}, {2, 2});
+  B = Slice::fromOffsetsSizes({2, 2}, {2, 2});
+  ASSERT_FALSE(A.doesOverlap(B));
+  ASSERT_FALSE(B.doesOverlap(A));
+}
+
+TEST_F(EventHistogramWriter, SliceOverfullOverlap) {
+  auto A = Slice::fromOffsetsSizes({1, 6}, {2, 2});
+  auto B = Slice::fromOffsetsSizes({0, 2}, {4, 8});
+  ASSERT_TRUE(A.doesOverlap(B));
+  ASSERT_TRUE(B.doesOverlap(A));
+}
+
+TEST_F(EventHistogramWriter, SlicePartialOverlap) {
+  auto A = Slice::fromOffsetsSizes({1, 6}, {2, 2});
+  auto B = Slice::fromOffsetsSizes({0, 7}, {2, 1});
+  ASSERT_TRUE(A.doesOverlap(B));
+  ASSERT_TRUE(B.doesOverlap(A));
+}
+
+json createTestWriterTypedJson() {
+  // The option 'convert_edge_type_to_float' is a temporary special feature for
```

Comment:
I would rather say that there's this extra conversion feature. Or at least say that is "required" (unless for some reason can/will be removed in future)

Reply:
I've created #304 and expanded the comment in the code.